guardian/cdk
# chore: Enable beta releases to NPM

## What does this change?


Whilst it is possible to install this package from GitHub, it can be useful to install from NPM as the published package structure varies slightly, for example in #675 .

With this change, we will be able to publish to the `beta` channel by update the `beta` branch. This will enable us to to install our experimental changes from NPM with:

```
npm install @guardian/cdk@beta
```

I followed [these docs](https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#prerelease).

## Does this change require changes to existing projects or CDK CLI?


No.

## Does this change require changes to the library documentation?


Yes and added.

## How to test



I don't think it's possible...

Once this PR is merged, we should be able to create a `beta` branch and observe it be published to `@guardian/cdk@beta`. We should also see updates to `main` continue to be released as normal.

## How can we measure success?


1. Updates to the `main` branch continue to be automatically released to the default NPM channel.
2. Updates to the `beta` branch will get automatically released to the beta NPM channel.

## Have we considered potential risks?


CD becomes a little more complicated with the added `if` statements. An alternative to this change would be to copy the `./.github/workflows/cd.yaml` and `./script/cd` files into `./.github/workflows/cd-beta.yaml` and `./script/cd-beta` and remove the `if` guards. I think this would create more maintenance, however.
